tools/extra/eosio_abigen_tool/eosio-abigen.cpp.in

```diff
@@ -240,6 +224,32 @@ class abigen : public generation_utils {
       _abi.tables.insert(t);
    }
 
+   void add_type( const clang::QualType& t ) {
+      clang::QualType type(t);
+      type = get_ignored_type(type);
+      if (!is_builtin_type(type)) {
+         if (is_aliasing(type))
+            add_typedef(type);
+         else if (is_template_specialization(type, {"vector", "set", "optional"})) {
+            if ( get_template_argument(type)->isRecordType() )
+               add_struct(get_template_argument(type).getTypePtr()->getAsCXXRecordDecl());
+            else
+               add_type(get_template_argument(type));
+         }
```

Comment:
I guess it's better to just call ```add_type``` in the ```vector/set/optional``` case,
```
         else if (is_template_specialization(type, {"vector", "set", "optional"})) {
               add_type(get_template_argument(type));
         }
```
This will fixes some types like: ```vector<string>```(vector of buildin), vector of map, etc.

Also I couldn't generate the correct abi for the following case: vector of a typedef of a primitive
```
typedef double dbltype;
struct [[eosio::table]] A {
   vector<dbltype> dbvec;
};
```
I guess we might need to remove "vector" in ```is_builtin_type( const clang::QualType& t )``` ?